homeassistant/components/songpal/config_flow.py

```diff
@@ -0,0 +1,158 @@
+"""Config flow to configure songpal component."""
+import logging
+from typing import Optional
+from urllib.parse import urlparse
+
+from songpal import Device
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_HOST, CONF_NAME
+
+from .const import CONF_ENDPOINT, CONF_MODEL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+DEFAULT_NAME = "Songpal"
+
+
+class SongpalConfig:
+    """Device Configuration."""
+
+    def __init__(self, name, host, endpoint, model=None):
+        """Initialize Configuration."""
+        self.name = name
+        self.host = host
+        self.endpoint = endpoint
+        self.model = model
+
+
+class SongpalConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Songpal configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the flow."""
+        self.conf: Optional[SongpalConfig] = None
+
+    async def _show_setup_form(self, user_input=None, errors=None):
+        user_input = user_input or {}
+        default_name = user_input.get(CONF_NAME) or DEFAULT_NAME
+        default_endpoint = user_input.get(CONF_ENDPOINT)
+        data_schema = {
+            vol.Optional(CONF_NAME, default=default_name): str,
+        }
+        if default_endpoint is not None:
+            data_schema.update(
+                {vol.Required(CONF_ENDPOINT, default=default_endpoint): str}
+            )
+        else:
+            data_schema.update({vol.Required(CONF_ENDPOINT): str})
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema(data_schema), errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return await self._show_setup_form()
```

Comment:
When a device is discovered, it should be automatically shown in the integration setup page. I can do something like that, but I don't think its necessary.

Reply:
Yes it will be shown indeed, if `ssdp` integration is initialized. During my first test I didn't have it on, that's why I got shown the form in the first place, I assume.